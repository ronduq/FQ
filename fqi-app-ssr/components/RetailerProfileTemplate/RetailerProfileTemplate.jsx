import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import get from 'lodash.get';
import s from './RetailerProfileTemplate.scss';
import layout from '../../hocs/Layout';
import AggregatedScore from '../AggregatedScore/AggregatedScoreContainer';
import CardList from '../CardList/CardListContainer';
import TableFilter from '../TableFilter/TableFilterContainer';
import RetailerProfileFilters from '../RetailerProfileFilters/RetailerProfileFiltersContainer';
import RetailerHighlightsFilter from '../RetailerHighlightsFilter/RetailerHighlightsFilterContainer';
import NotFoundDisplay from '../NotFoundDisplay/NotFoundDisplay';
import FilterToggle from '../FilterToggle/FilterToggleContainer';
import Tooltip from '../../hocs/TooltipContainer';
import SharePage from '../SharePage/SharePageContainer'

const RetailerProfileTemplate = ({
  compareBy,
  content,
  pageId,
  retailersLabels,
  selectedLocation,
  selectedRetailer,
  query,
  retailerRanks,
  notFoundScreens,
  viewport
}) => {
  const { label: selectedLocationLabel } = selectedLocation;

  const showSharePage = (show) => show && <SharePage 
    compareBy={compareBy}
    id={pageId} 
    includeRetailers
    includeProduces
    viewport={viewport} 
  /> 
  return (
    <div className={classNames('container', s.retailerProfile)}>
      <div className='layout layout--3-9 '>
        <div className='layout__column--1'>
          <FilterToggle>
            <RetailerProfileFilters currentRetailer={query.retailer} compareBy={compareBy} />
          </FilterToggle>
        </div>
        { retailerRanks && retailerRanks.length > 0 &&
          <div className='layout__column--2'>
            <section>

            <div className={s.columnHeader} >
              <Tooltip 
                side='right' 
                size='large'
                bubblePosition='right'
                id='retailer_title'
              >
                <h1 className={s.pageTitle}>{selectedRetailer} <span className={s.seperator}>in</span> <span className={s.location}>{selectedLocationLabel}</span></h1>
              </Tooltip>

                {showSharePage(viewport === 'xlarge')}
              </div>
              
              <AggregatedScore retailersLabels={retailersLabels} query={query} />
            </section>
            <section>
              <Tooltip 
                side='right' 
                size='large'
                bubblePosition='right'
                id='retailer_highlights'
              >
                <h2 className={classNames(s.pageTitle, s.highlights)}>{content.retailer_highlights_title}</h2>
              </Tooltip>  
              <RetailerHighlightsFilter compareBy={compareBy} content={content} />
              <CardList />
            </section>
            <section>
              <TableFilter />

              {showSharePage(viewport !== 'xlarge')}
            </section>
          </div>
        }
        {retailerRanks.length == 0 &&
          <div className='layout__column--2'>
            <h1 className={s.pageTitle}>{selectedRetailer} <span className={s.seperator}>in</span> <span className={s.location}>{selectedLocationLabel}</span></h1>
            <NotFoundDisplay 
              type='apple' 
              title={notFoundScreens[`retailer_profile`].title}
              text={`${notFoundScreens[`retailer_profile`].text} ${selectedRetailer}`} 
            />
          </div>
        }
      </div>
    </div>
  );
};

RetailerProfileTemplate.propTypes = {
  compareBy: PropTypes.string,
  content: PropTypes.shape(),
  retailersLabels: PropTypes.shape(),
  pageId: PropTypes.string,
  selectedLocation: PropTypes.shape(),
  selectedRetailer: PropTypes.string,
  query: PropTypes.shape(),
  viewport: PropTypes.string,
};

RetailerProfileTemplate.defaultProps = {
  content: {
    retailer_highlights_title: 'Retailer Highlights',
  },
  selectedLocation: {
    id: '',
    label: ''
  },
  query: {
    retailer: ''
  }
};

export default layout(RetailerProfileTemplate);