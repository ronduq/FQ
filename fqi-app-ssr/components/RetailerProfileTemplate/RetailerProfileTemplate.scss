@import '../../style/settings/colours';
@import '../../style/settings/fonts';
@import '../../style/settings/breakpoints';

.retailerProfile {

  @include respond-to('tablet', min-width) {
    margin-top: 35px;
  }

  .pageTitle {
    padding: 16px 0;
    text-align: center;
    @include font-size('medium');

    @include respond-to('tablet', min-width) {
      text-align: left;
      @include font-size('xlarge');
    }

    &.highlights {
      padding-top: 0;
    }
  }
  
  .columnHeader {
    display: flex;
    justify-content: space-between;
    padding: 0 16px;
  
    @include respond-to('tablet', min-width) {
      padding: 0;
    }
  }
}
