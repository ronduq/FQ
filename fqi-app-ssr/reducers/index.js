import { combineReducers } from 'redux';
import contentReducer from './content';
import retailersReducer from './retailers';
import producesReducer from './produces';
import locationsReducer from './locations';
import retailerProducesReducer from './retailerProduces';
import retailerRanksReducer from './retailerRanks';
import produceInfoReducer from './produceInfo';
import retailerTrendsReducer from './retailerTrends';
import produceTrendsReducer from './produceTrends';
import bestPicksReducer from './bestPicks';
import breadcrumbsReducer from './breadcrumbs';

const appReducer = () =>
  combineReducers({
    content: contentReducer,
    locations: locationsReducer,
    produces: producesReducer,
    retailers: retailersReducer,
    retailerRanks: retailerRanksReducer,
    retailerProduces: retailerProducesReducer,
    produceInfo: produceInfoReducer,
    retailerTrends: retailerTrendsReducer,
    produceTrends: produceTrendsReducer,
    bestPicks: bestPicksReducer,
    breadcrumbs: breadcrumbsReducer,
  });

// const rootReducer = history => (state, action) => {
//   if (action.type === 'LOGOUT') {
//     state = undefined;
//   }
//   return appReducer()(state, action);
// };

export default appReducer;
