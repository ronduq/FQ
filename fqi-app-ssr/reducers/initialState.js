export const initialRetailersState =  {
  name: 'Retailers',
  items: [],
  allItemsSelected: true,
  itemsLabels: {},
  itemsLogos: {},
  selectedRetailer: '',
}

export const initialProducesState =  {
  name: 'Produces',
  allItemsSelected: true,
  initialSelection: true,
  items: {},
  itemsImages: {},
  itemsLabels: {},
}

export const initialContentState = {
  contentFetched: false,
  footer: [],
  viewport: 'mobile',
  retailerTrends: {
    show_view_by_dropdown: false,
  },
  homepage: {},
}

export const initialLocationState = {
  locations: [],
  selectedLocation: {}
}

export const initialRetailerRanksState = {
  ranks: [],
  labels: {
    retailer: 'Retailer',
    quality: 'Quality Score',
    value: 'Value Rank',
    perception: 'Perception Rank',
    single_produce_average_title: 'Selected Produce Averagedddd'
  },
  ranksSingle: [],
  selected: 'quality',
  direction: 'topToBottom',
}

export const producesOfRetailer = {
  items: [],
  sortDirection: 'DSC',
  compareBy: 'quality',
  infoTable: {}
}

export const produceInfo = {
  info: {},
  labels: {},
}

export const initialRetailerTrendsState = {
  trends: [],
  viewBy: 'weekly',
  compareBy: 'quality',
  isLoading: false,
}

export const initialProduceTrendsState = {
  trends: [],
  viewBy: 'weekly',
  compareBy: 'quality',
  isLoading: false,
}

export const bestPicksState = {
  criteria: 'quality',
  topPicks: [],
  retailerTopPicks: []
}

export const breadcrumbs = {
  
}
