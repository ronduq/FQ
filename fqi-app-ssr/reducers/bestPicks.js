import update from 'react-addons-update';
import { bestPicksState } from './initialState';
import { 
  SET_BEST_PICKS,
  SORT_BEST_PICKS,
  SET_INITIAL_BEST_PICKS_CRITERIA
} from '../constants/'

const  bestPicksReducer = (state = bestPicksState, action) => {
  const { payload, type } = action;
  switch (type) {
    case SET_BEST_PICKS: {
      const { data } = payload;
      return update(state, {
        topPicks: {$set: data.topPicks},
        retailerTopPicks: {$set: data.retailerTopPicks},
      })
    }
    case SORT_BEST_PICKS: {
      const {criteria, direction} = payload;
      return update(state, {
        criteria: {$set: criteria},
      })
    }
    case SET_INITIAL_BEST_PICKS_CRITERIA: {
      const { criteria } = payload;
      return {
        ...state,
        criteria
      }
    }
    default:
      return state;
  }
};

export default  bestPicksReducer;
