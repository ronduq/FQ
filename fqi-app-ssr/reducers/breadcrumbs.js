import { breadcrumbs } from './initialState';
import { 
  SET_BREADCRUMBS,
} from '../constants/'

const breadcrumbsReducer = (state = breadcrumbs, action) => {
  const { payload, type } = action;
  switch (type) {
    case SET_BREADCRUMBS: {
      const { page, parentPage, grandparentPage } = payload;
      return {
        ...state,
        currentPage: page,
        parentPage,
        grandparentPage
      }
    }
    default:
      return state;
  }
};

export default breadcrumbsReducer;
