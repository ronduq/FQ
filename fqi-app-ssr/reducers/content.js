import { initialContentState } from './initialState';
import { arrayToObject } from '../utils';

import { 
    CONTENT_FETCHED,
    SET_FOOTER_CONTENT,
    SET_VIEWPORT,
    SET_PRODUCT_CARD_LABELS,
    SET_NOT_FOUND_SCREENS,
    SET_USDA_CONTENT,
    SET_RETAILER_TRENDS_CONTENT,
    SET_RETAILER_PROFILE_CONTENT,
    SET_PRODUCE_PROFILE_CONTENT,
    SET_HOMEPAGE_CONTENT,
    SET_HOMEPAGE_SLIDES_CONTENT,
    SET_PRODUCE_BASKET_CONTENT,
    SET_MATCH_CARDS_CONTENT,
    SET_BEST_PICKS_CONTENT,
    SET_PRODUCE_TRENDS_CONTENT,
    SET_FILTERS_CONTENT,
    SET_SUBSCRIBE_CONTENT,
    SET_TOOLTIPS_CONTENT,
    SET_SOCIAL_LINKS_CONTENT,
    SET_BREADCRUMBS_CONTENT,
    SET_HEADER_CONTENT,
    SET_OVERALL_RANK_CONTENT,
    SET_BUTTONS_CONTENT,
    SET_COOKIE_BANNER_CONTENT,
    SET_METADATA_CONTENT,
} from '../constants';

import { getViewportType } from '../utils'

const contentReducer = (state = initialContentState, action) => {
  const { type, payload } = action;
  switch (type) {
    case CONTENT_FETCHED: {
      return {
        ...state,
        contentFetched: true
      }
    }
    case SET_FOOTER_CONTENT: {
      const { footer } = payload;
      return {
        ...state,
        footer
      };
    }
    case SET_VIEWPORT: {
      const { width } = payload;
      return {
        ...state,
        viewport: getViewportType(width)
      }
    }
    case SET_PRODUCT_CARD_LABELS: {
      const { labels } = payload;
      return {
        ...state,
        productCardLabels: labels[0]
      }
    }
    case SET_NOT_FOUND_SCREENS: {
      const { labels } = payload;
      const notFoundScreens = {}
      labels.map((label) => {
        notFoundScreens[label.id] = label;
      });
      return {
        ...state,
        notFoundScreens: notFoundScreens
      }
    }
    case SET_USDA_CONTENT: {
      const { texts } = payload;
      return {
        ...state,
        usdaTexts: texts[0],
      }
    }
    case SET_RETAILER_TRENDS_CONTENT: {
      const { retailerTrends } = payload;
      return {
        ...state,
        retailerTrends: retailerTrends[0]
      }
    }
    case SET_RETAILER_PROFILE_CONTENT: {
      const { retailerProfile } = payload;
      return {
        ...state,
        retailerProfile: retailerProfile[0]
      }
    }
    case SET_PRODUCE_TRENDS_CONTENT: {
      const { produceTrends } = payload;
      return {
        ...state,
        produceTrends: produceTrends[0]
      }
    }
    case SET_PRODUCE_PROFILE_CONTENT: {
      const { produceProfile } = payload;
      return {
        ...state,
        produceProfile: produceProfile[0]
      }
    }
    case SET_HOMEPAGE_CONTENT: {
      const { homepage } = payload;
      return {
        ...state,
        homepage: homepage[0]
      }
    }
    case SET_HOMEPAGE_SLIDES_CONTENT: {
      const { labels } = payload;
      return {
        ...state,
        homepageSlides: labels,
      }
    }
    case SET_PRODUCE_BASKET_CONTENT: {
      const { basket } = payload;
      return {
        ...state,
        produceBasket: basket[0]
      }
    }
    case SET_MATCH_CARDS_CONTENT: {
      
      const { labels } = payload;
      return {
        ...state,
        matchCards: arrayToObject(labels, 'id')
      }
    }
    case SET_BEST_PICKS_CONTENT: {
      const { labels } = payload;
      return {
        ...state,
        bestPicks: labels[0]
      }
    }
    case SET_FILTERS_CONTENT: {
      const { labels } = payload;
      return {
        ...state,
        filtersContent: labels[0]
      }
    }
    case SET_SUBSCRIBE_CONTENT: {
      const { labels } = payload;
      return {
        ...state,
        subscribeContent: labels[0]
      }
    }
    case SET_TOOLTIPS_CONTENT: {
      const { labels } = payload;
      return {
        ...state,
        tooltips: labels
      }
    }
    case SET_SOCIAL_LINKS_CONTENT: {
      const { socialLinks } = payload;
      return {
        ...state,
        socialLinks: arrayToObject(socialLinks, 'id')
      }
    }
    case SET_BREADCRUMBS_CONTENT: {
      const { labels } = payload;
      let breadcrumbLabels = {};
      labels.map(label => {
        breadcrumbLabels[label.id] = label
      })
      return {
        ...state,
        breadcrumbs: breadcrumbLabels
      }
    }
    case SET_HEADER_CONTENT: {
      const { labels } = payload;
      return {
        ...state,
        header: labels[0]
      }
    }
    case SET_OVERALL_RANK_CONTENT: {
      const { labels } = payload;
      return {
        ...state,
        overallRank: labels[0]
      }
    }
    case SET_BUTTONS_CONTENT: {
      const { labels } = payload;
      return {
        ...state,
        buttons: labels[0]
      }
    }
    case SET_COOKIE_BANNER_CONTENT: {
      const { labels } = payload;
      return {
        ...state,
        cookieBanner: labels[0]
      }
    }

    case SET_METADATA_CONTENT: {
      const { labels } = payload;
      const metadataContent = {};
      labels.map(label => {
        metadataContent[label.id] = label
      });
      return {
        ...state,
        metadataContent: metadataContent
      }
    }

    

    default:
      return state;
  }
};

export default contentReducer;
