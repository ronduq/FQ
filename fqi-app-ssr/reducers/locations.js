import { initialLocationState } from './initialState';
import { 
  SET_LOCATIONS,
  SET_SELECTED_LOCATION,
  SET_CURRENT_PAGE,
} from '../constants/'

const locationsReducer = (state = initialLocationState, action) => {
  const { payload, type } = action;
  switch (type) {
    case SET_LOCATIONS: {
      const { locations } = payload;
      return {
        ...state,
        locations
      }
    }
    case SET_SELECTED_LOCATION: {
      const { selectedLocation } = payload;
      return {
        ...state,
        selectedLocation: state.locations.find(({id}) => id === selectedLocation)
      }
    }
    case SET_CURRENT_PAGE: {
      const{ currentPage, metaExtra } = payload;
      return {
        ...state,
        currentPage,
        metaExtra,
      }
    }
    default:
      return state;
  }
};

export default locationsReducer;
