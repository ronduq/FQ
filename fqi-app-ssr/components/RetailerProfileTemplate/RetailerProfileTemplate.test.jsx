/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';

import layout from '../../hocs/Layout';
import { initialRetailerRanksState, initialRetailersState, producesOfRetailer, initialProducesState, initialContentState } from '../../reducers/initialState'
import RetailerProfileTemplate from './RetailerProfileTemplate';

jest.mock('../../hocs/Layout', () => ({
  __esModule: true,
  default: (e) => e,
}));

const fakeStore = {
  dispatch: jest.fn(),
  getState: () => ({
    content: initialContentState,
    produces: initialProducesState,
    retailers: initialRetailersState,
    retailerRanks: initialRetailerRanksState,
    retailerProduces: producesOfRetailer,
    content: {
      viewport: 'mobile'
    },
    locations:{
      selectedLocation: {
        id: 'boston',
        label: 'Boston, MA'
      }
    },
  }),
  subscribe: jest.fn()
};

describe('RetailerProfileTemplate', () => {

  const props = {
    content: {
      retailer_highlights_title: 'Retailer Highlights'
    },
    retailersLabels:{
      single_produce_average_title: 'Single produce',
      quality: 'quality',
      value: 'value',
      perception: 'perception',
    },
    query: {
      location: 'boston',
      retailer: 'aldi'
    },
    selectedLocation: { label: 'Boston', id: 'boston'},
    selectedRetailer: 'aldi',
    retailerProduces: producesOfRetailer,
    retailersData: initialRetailersState,
    producesData: {
      allItemsSelected: true,
      items: {
        '1': {
          id: "apple-braeburn",
          isChild: true,
          isParent: false,
          parentCode: "apple",
          produceCode: "apple-braeburn",
          produceName: "Apple Braeburn",
          selected: true,
        },
        '2': {
          id: "apple-fuji",
          isChild: true,
          isParent: false,
          parentCode: "apple",
          produceCode: "apple-fuji",
          produceName: "Apple Fuji",
          selected: true,
        }
      },
      itemsImages: {
        'apple-braeburn': "https://cdn.buttercms.com/ZIjnt41SVemFWkksYchT", 
        'apple-fuji': "https://cdn.buttercms.com/DP5q6Yb5Q8SQbYsWdDeJ"
      },
      itemsLabels: {
        'apple-braeburn': "Braeburn Apples", 
        'apple-fuji': "Fuji Apples", 
      },
      name: "Produces",
    },
    cardLabels: {
      countries_of_origin: "Countries of origin",
      last_scan: "Last scan",
      organic: "Organic",
      perception: "Perception",
      product_of: "Product of",
      profile: "Profile",
      quality: "Quality",
      trends: "Trends",
      value: "Value",
    },
    retailerRanksData: initialRetailerRanksState,
    retailerRanks: [],
    notFoundScreens: {
      retailer_profile: {
        title: 'test',
        text: 'text'
      }
    }
  };

  it('RetailerProfileTemplate snapshot"', () => {

    const component = renderer.create(
    <Provider store={fakeStore}>
      <RetailerProfileTemplate {...props} />
    </Provider>);
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  });

})