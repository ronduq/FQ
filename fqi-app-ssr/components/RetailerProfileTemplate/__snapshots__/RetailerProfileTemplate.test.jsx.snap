// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetailerProfileTemplate RetailerProfileTemplate snapshot" 1`] = `
<div
  className="container retailerProfile"
>
  <div
    className="layout layout--3-9 "
  >
    <div
      className="layout__column--1"
    >
      <div
        className="filterToggle"
      >
        <button
          aria-controls="FilterMenu"
          aria-expanded={false}
          aria-haspopup="true"
          aria-label="Toggle Filters"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="toggleIcon"
            focusable="false"
          >
            <use
              xlinkHref="#mobileFilter"
            />
          </svg>
        </button>
        <div
          className="twoColumn"
          id="FilterMenu"
        />
      </div>
    </div>
    <div
      className="layout__column--2"
    >
      <h1
        className="pageTitle"
      >
        aldi
         
        <span
          className="seperator"
        >
          in
        </span>
         
        <span
          className="location"
        >
          Boston
        </span>
      </h1>
      <div
        className="notFoundDisplay"
      >
        <div
          className="notFoundDisplayImg"
        >
          <svg
            aria-hidden="true"
            className="icon iconHuge iconTeal"
            focusable="false"
          >
            <use
              xlinkHref="#apple"
            />
          </svg>
        </div>
        <p
          className="notFoundDisplayTitle"
        >
          test
        </p>
        <p
          className="notFoundDisplayText"
        >
          text aldi
        </p>
      </div>
    </div>
  </div>
</div>
`;
